default/scripting/buildings/EXPERIMENTOR_OUTPOST.focs.py

```diff
@@ -0,0 +1,302 @@
+)
+
+# the following modifies Experimentor spawn start by a factor of 0.1 up for low planet density, and 0.1 down for high planet density
+# for galaxy sizes above 200 increases it somewhat as the galaxy size grows
+EXPERIMENTOR_SPAWN_START_TURN = (
+    EXPERIMENTOR_SPAWN_CALC_A * (1.2 - GalaxyPlanetDensity / 10) * (MaxOf(float, 1, GalaxySize / 200)) ** 0.4
+)
+
+# of the 5 closest systems to which a lane could be added, pick one
+EXPERIMENTOR_ADD_STARLANE = AddStarlanes(
+    endpoint=NumberOf(
+        number=1,
+        condition=MinimumNumberOf(
+            number=5,
+            sortkey=DirectDistanceBetween(target=Source.ID, source=LocalCandidate.ID),
+            condition=System & ~Contains(IsSource),
+        ),
+    ),
+    condition=CAN_ADD_STARLANE_TO_SOURCE,
+)
+
+BuildingType(  # type: ignore[reportUnboundVariable]
+    name="BLD_EXPERIMENTOR_OUTPOST",
+    description="BLD_EXPERIMENTOR_OUTPOST_DESC",
+    buildcost=1,
+    buildtime=1,
+    location=IsAnyObject,
+    effectsgroups=[
+        EffectsGroup(
+            scope=IsSource | (Object(id=Source.PlanetID) & Planet()),
+            effects=[SetStealth(value=Value + 60)],
+        ),
+        EffectsGroup(
+            scope=(Object(id=Source.SystemID) | (InSystem(id=Source.SystemID) & Planet() & IsSource)),
+            effects=[SetStealth(value=Value + 40)],
+        ),
+        EffectsGroup(
+            scope=Object(id=Source.PlanetID),
+            effects=[
+                SetMaxShield(value=Value + (60000 * PLANET_SHIELD_FACTOR)),
+                SetMaxDefense(value=Value + (EXPERIMENTOR_SPAWN_START_TURN * PLANET_DEFENSE_FACTOR)),
+                SetMaxTroops(value=Value + 300),
+                SetDetection(value=Value + 100),
+            ],
+        ),
+        EffectsGroup(
+            scope=Object(id=Source.PlanetID),
+            activation=(ContainedBy(Contains(Ship & OwnedBy(affiliation=AnyEmpire)))),
+            effects=[
+                # Regeneration
+                SetDefense(value=Value + (10 * PLANET_DEFENSE_FACTOR)),
+                SetShield(value=Value + (50 * PLANET_SHIELD_FACTOR)),
+            ],
+        ),
+        EffectsGroup(
+            scope=(IsSource & OwnedBy(affiliation=AnyEmpire) & HasSpecies(name=["SP_EXPERIMENTOR"])),
+            effects=[
+                Victory(reason="VICTORY_EXPERIMENTOR_CAPTURE"),
+                EXPERIMENTOR_ADD_STARLANE,
+                EXPERIMENTOR_ADD_STARLANE,
+                EXPERIMENTOR_ADD_STARLANE,
+                EXPERIMENTOR_ADD_STARLANE,
+                EXPERIMENTOR_ADD_STARLANE,
+                Destroy,
+            ],
+        ),
+        EffectsGroup(
+            scope=IsSource & OwnedBy(affiliation=AnyEmpire),
+            activation=(
+                # always remove lanes before spawn start turn
+                Turn(high=EXPERIMENTOR_SPAWN_START_TURN)
+                | ContainedBy(Contains(Monster & Unowned & ~Design(name="SM_EXP_OUTPOST")))
```

Comment:
Maybe Convert Design to IsDesign?

Reply:
Or `HasDesign`? @geoffthemedio

Reply:
I prefer `HasDesign`

Reply:
@Cjkjvfnby It's landed in parser as `HasDesign`